truffle/src/main/ruby/core/proc.rb

```diff
@@ -33,18 +41,31 @@ class Proc
 
   def curry(curried_arity = nil)
     if lambda? && curried_arity
+
       if arity >= 0 && curried_arity != arity
-        raise ArgumentError, "Wrong number of arguments (%i for %i)" % [
-          curried_arity,
-          arity
-        ]
+        raise ArgumentError, "Wrong number of arguments (#{curried_arity} for #{arity})"
```

Comment:
I've also now split other `ArgumentError` strings across two lines, mostly to make them more readable with the inline variables.